[addons] allow on multiple selectable instance support one as default

AlwinEsch opened the pull request:
## Description

With this change, activating an add-on that allows multiple instances via settings will automatically add one.

In addition, the add-on is deactivated again when the last instance is removed in settings.

Further becomes with this a crash fixed, if an user removed all instances in settings, leaved add-on as enabled and then goes inside Kodi's main menu to "TV" field.
```gdb
Thread 1 "kodi.bin" received signal SIGSEGV, Segmentation fault.
0x00007ffff6446994 in pthread_mutex_lock () from /usr/lib/libc.so.6
(gdb) bt
#0  0x00007ffff6446994 in pthread_mutex_lock () at /usr/lib/libc.so.6
#1  0x0000555556d280f1 in PVR::CPVRProvider::GetIconPath[abi:cxx11]() const ()
#2  0x0000555556d3736a in PVR::CPVRGUIInfo::GetListItemAndPlayerLabel(CFileItem const*, KODI::GUILIB::GUIINFO::CGUIInfo const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) const ()
#3  0x0000555556d37ed4 in PVR::CPVRGUIInfo::GetLabel(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, CFileItem const*, int, KODI::GUILIB::GUIINFO::CGUIInfo const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*) const ()
#4  0x00005555567feb82 in KODI::GUILIB::GUIINFO::CGUIInfoProviders::GetLabel(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, CFileItem const*, int, KODI::GUILIB::GUIINFO::CGUIInfo const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*) const ()
#5  0x0000555556b888da in CGUIInfoManager::GetMultiInfoItemLabel(CFileItem const*, int, KODI::GUILIB::GUIINFO::CGUIInfo const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*) const ()
#6  0x0000555556b8938f in CGUIInfoManager::GetMultiInfoItemImage(CFileItem const*, int, KODI::GUILIB::GUIINFO::CGUIInfo const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*) const ()
#7  0x0000555556b89556 in CGUIInfoManager::GetItemImage(CGUIListItem const*, int, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*) const ()
#8  0x0000555556802fd5 in KODI::GUILIB::GUIINFO::CGUIInfoLabel::GetItemLabel(CGUIListItem const*, bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*) const ()
#9  0x0000555556879cc0 in CGUIImage::UpdateInfo(CGUIListItem const*) ()
#10 0x0000555556889e23 in CGUIListGroup::UpdateInfo(CGUIListItem const*) ()
#11 0x0000555556889e23 in CGUIListGroup::UpdateInfo(CGUIListItem const*) ()
#12 0x00005555568925b5 in CGUIListItemLayout::Process(CGUIListItem*, int, unsigned int, std::vector<CDirtyRegion, std::allocator<CDirtyRegion> >&) ()
#13 0x0000555556837a9b in CGUIBaseContainer::ProcessItem(float, float, std::shared_ptr<CGUIListItem>&, bool, unsigned int, std::vector<CDirtyRegion, std::allocator<CDirtyRegion> >&) ()
#14 0x000055555689c11f in CGUIPanelContainer::Process(unsigned int, std::vector<CDirtyRegion, std::allocator<CDirtyRegion> >&) ()
#15 0x0000555556845f1b in CGUIControl::DoProcess(unsigned int, std::vector<CDirtyRegion, std::allocator<CDirtyRegion> >&) ()
#16 0x0000555556830139 in CGUIBaseContainer::DoProcess(unsigned int, std::vector<CDirtyRegion, std::allocator<CDirtyRegion> >&) ()
#17 0x000055555685792c in CGUIControlGroupList::Process(unsigned int, std::vector<CDirtyRegion, std::allocator<CDirtyRegion> >&) ()
#18 0x0000555556845f1b in CGUIControl::DoProcess(unsigned int, std::vector<CDirtyRegion, std::allocator<CDirtyRegion> >&) ()
#19 0x0000555556854082 in CGUIControlGroup::Process(unsigned int, std::vector<CDirtyRegion, std::allocator<CDirtyRegion> >&) ()
#20 0x0000555556845f1b in CGUIControl::DoProcess(unsigned int, std::vector<CDirtyRegion, std::allocator<CDirtyRegion> >&) ()
#21 0x0000555556854082 in CGUIControlGroup::Process(unsigned int, std::vector<CDirtyRegion, std::allocator<CDirtyRegion> >&) ()
#22 0x0000555556845f1b in CGUIControl::DoProcess(unsigned int, std::vector<CDirtyRegion, std::allocator<CDirtyRegion> >&) ()
#23 0x0000555556854082 in CGUIControlGroup::Process(unsigned int, std::vector<CDirtyRegion, std::allocator<CDirtyRegion> >&) ()
#24 0x0000555556845f1b in CGUIControl::DoProcess(unsigned int, std::vector<CDirtyRegion, std::allocator<CDirtyRegion> >&) ()
#25 0x0000555556854082 in CGUIControlGroup::Process(unsigned int, std::vector<CDirtyRegion, std::allocator<CDirtyRegion> >&) ()
#26 0x0000555556845f1b in CGUIControl::DoProcess(unsigned int, std::vector<CDirtyRegion, std::allocator<CDirtyRegion> >&) ()
#27 0x00005555568c973d in CGUIWindow::DoProcess(unsigned int, std::vector<CDirtyRegion, std::allocator<CDirtyRegion> >&) ()
#28 0x00005555568cce29 in CGUIWindowManager::Process(unsigned int) ()
#29 0x0000555556b224ff in CApplication::FrameMove(bool, bool) ()
#30 0x0000555556b19605 in CApplication::Run() ()
#31 0x000055555677831e in XBMC_Run ()
#32 0x000055555602f5a8 in main ()
(gdb) 
```

<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here. -->

## Motivation and context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## How has this been tested?
<!--- Please describe in detail how you tested your change. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## What is the effect on users?
<!--- Summarize the effect of this change on Kodi end-users. -->
<!--- If the PR does not have a noticeable impact (e.g., if it only changes documentation), -->
<!--- just leave it empty. Put in more detail the bigger the impact is. -->
<!--- This section may be used for automatic creation of release notes. -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

ksooo commented on addons/resource.language.en_gb/resources/strings.po:
```diff
@@ -4979,7 +4979,11 @@ msgctxt "#10022"
 msgstr ""
 
-#empty string with id 10023
+#. Dialog label used in add-on setup about user inform by remove of last configuration
```
Dialog label used in add-on setup dialog to inform the user about removal of last add-on configuration

ksooo commented on addons/resource.language.en_gb/resources/strings.po:
```diff
@@ -4979,7 +4979,11 @@ msgctxt "#10022"
+#. Dialog label used in add-on setup about user inform by remove of last configuration
+#: xbmc/addons/gui/GUIDialogAddonSettings.cpp
+msgctxt "#10023"
+msgid "Add-on becomed disabled by remove of last configuration."
```
Add-on was disabled because last configuration was removed.

ksooo commented on xbmc/addons/IAddon.h:
```diff
@@ -31,6 +31,11 @@ constexpr const char* ADDON_SETTING_INSTANCE_ENABLED_VALUE = "kodi_addon_instanc
+/*!
+ * @brief Identifier denoting initial first add-on instance.
+ */
+constexpr AddonInstanceId ADDON_INSTANCE_BEGIN_ID = 1;
```
ADDON_FIRST_INSTANCE_ID

ksooo commented:
@AlwinEsch as discussed on Slack, instead of disabling the add-on after removal of last instance, we should simply forbid to remove the last instance. => Hide "Remove add-on configuration" item if only one configuration is left.

AlwinEsch committed:
[addons] allow on multiple selectable instance support one as default

With this change, activating an add-on that allows multiple instances via settings will automatically add one.

Also must always one instance present.

AlwinEsch commented:
Is updated and not allow remove if only one instance is used.

AlwinEsch committed:
[pvr] not use instance id to get client id inside first running one

As background to allow backward compatibility about database, becomes with this change the add of instance id to create pvr client id not used.
This allows to use already defined values of older add-ons inside an new one.

ksooo commented:
FreeBSD Jenkins error is unrelated.
